python/tk_desktop/project_communication.py

```diff
@@ -29,13 +34,21 @@ def __init__(self, engine):
 
     def connect_to_server(self, pipe, auth, disconnect_callback):
         """
-        Sets up a server to communicate with the background process and connects to the site engine.
+        Sets up a server to communicate with the foreground process and connects
+        back to the site engine.
         """
         # create the connection to the site engine.
         self._create_proxy(pipe, auth)
 
-        # register our side of the pipe as the current app proxy
-        self._create_server()
+        # register our side of the pipe as the current app proxy.
+        # Based on how the server expects us to connect to them, we'll
+        # use the right server to listen to their requests.
+        # This boils down to if the server suggested we use multiprocessing
+        # to connect back to them, we'll use that to receive their message
+        # since it's likely they are using that on their end as well.
+        # If the suggested we use http to connect back to them, then we'll listen
```

Comment:
I think this sentence is missing something, it doesn't quite read correctly.

Reply:
Also if I understand this correctly, both the project side and the site side are registering servers, so that both can receive instructions.

Based on that understanding, I'm not sure the comment is clear:
> Based on how the server expects us to connect to them, we'll use the right server to listen to their requests.

Which server? I think this should maybe be:
"Based on how the site server expects us to connect to them, we'll use the right server to listen to their requests."

Reply:
I rewrote the blurb, let me know what you think!

Reply:
Yeah I think that makes it clearer. But since this is in the `ProjectComunication` class, I still think it doesn't hurt to say site server when talking about the server's expectations, since this isn't a generic class that could be used on either side of the conversation.

> If the suggested we use http to connect back to them, then we'll listen on that as well.

as well or instead? I think instead?
Sorry being a bit picky, I think it looks better though!